### ISSUE TITLE
Intercept crashes when casting a nil to game_value

### ISSUE THREAD

**Issue by esteldunedain** [NouberNou/intercept#64]
My code works properly with an object returned by `sqf::near_object` for 4 commands. But after that it systematically crashes here:

![image](https://cloud.githubusercontent.com/assets/7674951/12698264/2ff493ae-c777-11e5-8af4-513eb234d173.png)

As far as I understand the object might be freed to early; maybe a problem with reference counters?

The code is this:
https://github.com/acemod/ACE3/tree/interceptNametags/extensions/nametags

And the line in which it alway fails is this:
https://github.com/acemod/ACE3/blob/interceptNametags/extensions/nametags/nametagger.cpp#L112

All the returned results look correct up to that point.

I tried to make a minimal repro case but failed so far. I'll keep trying.

**Comment by NouberNou** [comment]
Can you try not using any `get_variable` calls and see if it still crashes?

**Comment by esteldunedain** [comment]
I will

**Comment by esteldunedain** [comment]
@NouberNou, I removed all the calls to `get_variable` in this branch:

https://github.com/acemod/ACE3/tree/interceptNametagsNoGetVariable/extensions/nametags

with this commit: https://github.com/acemod/ACE3/commit/08b721dd2c3c4c79c2d84d88b8a64c317e098c54

The error is still happening in the exact same way, on the exact same line.

Anything else I can do to help? Given the CTD happens on the first call to the `on_frame` handler it would be pretty straightforward to monitor what's happening to the memory where the obejct is stored. I don't know how to go about that though.

**Comment by esteldunedain** [comment]
I think I've found the problem: `assignedTeam` is trying to return `nil` (which happens in some cases according to the [documentation][(https://community.bistudio.com/wiki/assignedTeam). That is casted to `game_value` in the `__string_unary_object` function.  The constructor is here:

https://github.com/NouberNou/intercept/blob/master/src/client/intercept/shared/types.cpp#L420-L425

![image](https://cloud.githubusercontent.com/assets/7674951/12732699/864f9c86-c915-11e5-95de-b9aa03c17a69.png)

Notice how `internal_.data = 0x0000`, hence trying to increment the ref count fails.

I think the game_value constructors should test for this before trying to increment the ref count.

**Comment by esteldunedain** [comment]
Confirmed, other function calls crash in the exact same way when they return nil. e.g.:
```c++
    object player(sqf::player);
    game_value myValue(sqf::get_hit_point_damage(player, "myNonExistingHitPoint"));
```

**Comment by esteldunedain** [comment]
Fixed in aa9bdab34fd6aedc7c711139417e4a40661b722b
